src/_components/AddAsset/AddAssetContainer.jsx

```diff
@@ -0,0 +1,173 @@
+        filteredModelNumbers: [],
+        modelNumber: 0,
+        assets: nextProps.assets
+      }
+    }
+    return null;
+  }
+
+  handleDropdownChanges = (event, data) => {
+    const { name, value } = data;
+    const { subcategories, assetTypes, assetMakes, modelNumbers } = this.props;
+
+    if (name === 'asset-category') {
+      this.setState({
+        filteredSubCategories: filterSubCategories(subcategories, value)
+      });
+    } else if (name === 'asset-subcategory') {
+      this.setState({
+        filteredAssetTypes: filterAssetTypes(assetTypes, value)
+      });
+    } else if (name === 'asset-types') {
+      this.setState({
+        filteredAssetMakes: filterAssetMakes(assetMakes, value)
+      });
+    } else if (name === 'asset-makes') {
+      this.setState({
+        filteredModelNumbers: filterModelNumbers(modelNumbers, value)
+      });
+    }
+  };
+
+  onSelectModelNumber = (event, data) => {
+    this.setState({ modelNumber: data.value });
+  };
+
+  onAddSerialNumber = (event) => {
+    this.setState({ serialNumber: event.target.value });
+  };
+
+  onAddAssetTag = (event) => {
+    this.setState({ assetTag: event.target.value });
+  };
+
+  onCreateAsset = (event, data) => {
+    this.props.createAsset({
+      "asset_code": this.state.assetTag,
+      "serial_number": this.state.serialNumber,
+      "model_number": this.state.modelNumber
+    });
+    event.target.reset();
+  };
+
+  render() {
+    return (
+      <AddAssetComponent {...this.props}
+                         handleDropdownChanges={this.handleDropdownChanges}
+                         onSelectModelNumber={this.onSelectModelNumber}
+                         onAddSerialNumber={this.onAddSerialNumber}
+                         onAddAssetTag={this.onAddAssetTag}
+                         onCreateAsset={this.onCreateAsset}
+                         filteredSubCategories={this.state.filteredSubCategories}
+                         filteredAssetTypes={this.state.filteredAssetTypes}
+                         filteredAssetMakes={this.state.filteredAssetMakes}
+                         filteredModelNumbers={this.state.filteredModelNumbers}
+                         modelNumber={this.state.modelNumber}
+                         serialNumber={this.state.serialNumber}
+                         assetTag={this.state.assetTag}
+      />
+    );
+  }
+}
+
+AddAssetContainer.propTypes = {
+  categoriesList: PropTypes.array,
+  subcategoriesList: PropTypes.array,
+  assetTypesList: PropTypes.array,
+  assetMakesList: PropTypes.array,
+  modelNumbersList: PropTypes.array,
+  assetsList: PropTypes.array,
+  loadCategories: PropTypes.func.isRequired,
+  loadSubCategories: PropTypes.func.isRequired,
+  loadAssetTypes: PropTypes.func.isRequired,
+  loadAssetMakes: PropTypes.func.isRequired,
+  loadModelNumbers: PropTypes.func.isRequired,
+  createAsset: PropTypes.func.isRequired
+};
+
+const mapStateToProps = ({
+                           categoriesList,
```

Comment:
The destructure state should be indented 2 spaces from the start of the line. Currently the indentation is off. I.e.
```
const mapStateToProps = ({
  categoriesList,
  subcategoriesList,
  ...
  assetsList
}) => ({
  categories: categoriesList,
  subcategories: subcategoriesList,
  ...
  assets: assetsList
});
```

Reply:
Noted. I have fixed this.